pages/glossary/programatically-determined-link-context.md

```diff
@@ -0,0 +1,20 @@
+--
+title: Programmatically Determined Link Context
+key: programmatically-determined-link-context
+--
+
+In HTML, information that is programmatically determinable from a link includes [text nodes][] that are [included in the accessibility tree][] and are
+
+- sibling [phrasing content][] nodes in the same [paragraph][] as the link;
+- sibling [phrasing content][] nodes in the same list (HTML element with the semantic role of `listitem`) as the link;
+- sibling [phrasing content][] nodes in the same table cell (HTML element with the semantic role of `cell` or `gridcell`) as the link;
```

Comment:
What if a link is contained inside a list inside a paragraph inside a table cell? Do you get to use all of them as context? I would argue it only counts towards the closest one.

Reply:
I don't agree. I think all should be used as context. In the example below don't you think that the table header and the text in the same cell are providing context even if the element is inside a list in the cell?

```html
<table>
  <tr>
    <th colspan="3">Ulysses</th>
  </tr>
  <tr>
    <td>
      Download the book in your preferred format:
      <ul>
        <li>
          <a href="https://www.gutenberg.org/files/4300/4300-h/4300-h.htm">
            HTML
          </a>
        </li>
        <li>
          <a
            href="https://www.gutenberg.org/ebooks/4300.epub.images?session_id=04cd710372888de8d8d322215cdfe8ce5b0f8d73"
          >
            EPUB
          </a>
        </li>
        <li>
          <a href="https://www.gutenberg.org/files/4300/4300-0.txt">
            Plain text
          </a>
        </li>
      </ul>
    </td>
  </tr>
</table>
```